storages/backends/gcloud.py

```diff
@@ -142,12 +146,19 @@ def get_default_settings(self):
             # roll over.
             "max_memory_size": setting("GS_MAX_MEMORY_SIZE", 0),
             "blob_chunk_size": setting("GS_BLOB_CHUNK_SIZE"),
+            "sa_email": setting("GS_SA_SIGNING_EMAIL")
         }
 
     @property
     def client(self):
+        credentials, project_id = auth.default(scopes=['https://www.googleapis.com/auth/cloud-platform'])
```

Comment:
This now completely eliminates `self.project_id` as a setting, is that always valid?

Reply:
Yes, otherwise the credentials might not match the project your request from.

Reply:
Mostly I'm asking about backwards compat here.

Reply:
Should work fine. Basically you fetch creds from gcp server based on google auth patterns.

Reply:
auth.default does not accept project anyways

Reply:
Don't suppose you have any documentation link handy? Just trying to think through all the ramifications other than the obvious one that we are adding a network call.

You can see that as originally written we cached the client. With this update we do the work but don't update the cached item, which is inconsistent.

Reply:
I can do a check if its empty, and if its provided we can use it. I don't want to break backwards compat.

Reply:
So if its not provided now it will use the one from the envs

Reply:
> Don't suppose you have any documentation link handy? Just trying to think through all the ramifications other than the obvious one that we are adding a network call.

The docs for this were a mess. I had to read source. Will link.

Reply:
https://github.com/googleapis/google-auth-library-python/blob/main/google/auth/_default.py#L541

Reply:
https://github.com/googleapis/google-auth-library-python/blob/main/google/auth/_default.py#L575 <- this is the relevant line